core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -840,7 +852,7 @@ class LogManager(logDirs: Seq[File],
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
       removedLog.renameDir(Log.logDeleteDirName(topicPartition))
-      checkpointLogRecoveryOffsetsInDir(removedLog.dir.getParentFile)
+      checkpointLogRecoveryOffsetsInDir(removedLog.dir.getParentFile, Some(Seq(removedLog)))
```

Comment:
Similar to the above, since removedLog will be deleted soon, there is no need to clean the snapshot for this partition. We could just pass in None.

Reply:
Ditto. I think it'd better pass it explicitly since only passing None could also means doing for all logs. In `checkpointLogRecoveryOffsetsInDir`, all logs ending with `-delete` will be excluded.